Library/Homebrew/version.rb

```diff
@@ -379,3 +392,29 @@ def self._parse(spec)
     return m.captures.first unless m.nil?
   end
 end
+
+class HeadVersion < Version
+  attr_reader :commit
+
+  def initialize(val)
+    unless val.respond_to?(:to_str)
```

Comment:
Since `HeadVersion` new is only used by `Version.create`, which already does the type-checking, is this needed?